infobip_channels/core/channel.py

```diff
@@ -135,11 +135,17 @@ def _construct_response(
             response_class = self._get_custom_response_class(
                 raw_response, *args, **kwargs
             )
+
+            if type(raw_response.json()) is list:
```

Comment:
You can call raw_response.json() once before the if and store it in a variable so we don't call it three times.

```python
response_json = raw_response.json()
if type(response_json) is list:
    raw_response_data = {"list": response_json}
else:
    raw_response_data = response_json
```

Reply:
👍🏻